client.messages.swarm.Task - slot should be marked @Nullable

## Description

I am trying to get a list of service tasks for a "global" mode swarm service.
global mode services do not have task slots - there is one task per host.

see https://github.com/docker/docker/blob/master/api/types/swarm/task.go#L45

## How to reproduce

create a global service in a docker swarm mode cluster and try to get a list of the service's tasks.
`service = client.listTasks(Task.find().serviceName(SERVICE_NAME).build());`

## What do you expect

the list of tasks for the service is returned

## What happened instead

an exception is thrown:
```
com.spotify.docker.client.exceptions.DockerRequestException: Request error: GET http://10.9.1.237:2375/tasks?filters=%7B%22service%22%3A%5B%22zuul-proxy%22%5D%7D: 200
	at com.spotify.docker.client.DefaultDockerClient.propagate(DefaultDockerClient.java:2163)
	at com.spotify.docker.client.DefaultDockerClient.request(DefaultDockerClient.java:2092)
	at com.spotify.docker.client.DefaultDockerClient.listTasks(DefaultDockerClient.java:1778)
	at com.goodstag.docker.lb.Application$DockerListener.getSwarmModeFilteredHosts(Application.java:292)
	at com.goodstag.docker.lb.Application$DockerListener.doProcessAll(Application.java:212)
	at com.goodstag.docker.lb.Application$DockerListener.run(Application.java:177)
	at java.lang.Thread.run(Thread.java:745)
Caused by: javax.ws.rs.client.ResponseProcessingException: com.fasterxml.jackson.databind.JsonMappingException: Can not construct instance of com.spotify.docker.client.messages.swarm.Task, problem: Null slot
 at [Source: org.glassfish.jersey.message.internal.ReaderInterceptorExecutor$UnCloseableInputStream@d4bb715; line: 1, column: 2096] (through reference chain: java.util.ArrayList[0])
```

## Software:

- docker version: 
```
Client:
 Version:      1.13.0
 API version:  1.25
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:44:08 2017
 OS/Arch:      linux/amd64

Server:
 Version:      1.13.0
 API version:  1.25 (minimum version 1.12)
 Go version:   go1.7.3
 Git commit:   49bf474
 Built:        Tue Jan 17 09:44:08 2017
 OS/Arch:      linux/amd64
 Experimental: false
```

- Spotify's docker-client version: 
`8.1.0`

--- a/src/main/java/com/spotify/docker/client/messages/swarm/GlobalService.java
+++ b/src/main/java/com/spotify/docker/client/messages/swarm/GlobalService.java
@@ -24,6 +24,7 @@ import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.ANY;
 import static com.fasterxml.jackson.annotation.JsonAutoDetect.Visibility.NONE;
 
 import com.fasterxml.jackson.annotation.JsonAutoDetect;
+import com.fasterxml.jackson.annotation.JsonCreator;
 import com.google.auto.value.AutoValue;
 
 @AutoValue
@@ -39,6 +40,11 @@ public abstract class GlobalService {
 
     public abstract GlobalService build();
 
+  }
+
+  @JsonCreator
+  static GlobalService create() {
+    return builder().build();
   }
 
 }

--- a/src/main/java/com/spotify/docker/client/messages/swarm/Task.java
+++ b/src/main/java/com/spotify/docker/client/messages/swarm/Task.java
@@ -65,6 +65,7 @@ public abstract class Task {
   @JsonProperty("ServiceID")
   public abstract String serviceId();
 
+  @Nullable
   @JsonProperty("Slot")
   public abstract Integer slot();
 

--- a/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
+++ b/src/test/java/com/spotify/docker/client/DefaultDockerClientTest.java
@@ -4475,6 +4475,29 @@ public class DefaultDockerClientTest {
   }
 
   @Test
+  public void testListTasksWithGlobalService() throws Exception {
+    requireDockerApiVersionAtLeast("1.24", "swarm support");
+    final int startingNumTasks = sut.listTasks().size();
+
+    final TaskSpec taskSpec = TaskSpec.builder()
+        .containerSpec(ContainerSpec.builder().image("alpine")
+            .command(new String[] { "ping", "-c1000", "localhost" }).build())
+        .build();
+
+    final ServiceSpec spec = ServiceSpec.builder()
+        .name(randomName())
+        .taskTemplate(taskSpec)
+        .mode(ServiceMode.withGlobal())
+        .build();
+
+    final ServiceCreateResponse response = sut.createService(spec);
+    assertThat(response.id(), is(notNullValue()));
+
+    await().until(numberOfTasks(sut), is(greaterThan(startingNumTasks)));
+    sut.listTasks();
+  }
+
+  @Test
   public void testListTaskWithCriteria() throws Exception {
     requireDockerApiVersionAtLeast("1.24", "swarm support");
 
@@ -4598,6 +4621,14 @@ public class DefaultDockerClientTest {
       public Boolean call() throws Exception {
         final ContainerInfo containerInfo = client.inspectContainer(containerId);
         return containerInfo.state().running();
+      }
+    };
+  }
+
+  private Callable<Integer> numberOfTasks(final DockerClient client) {
+    return new Callable<Integer>() {
+      public Integer call() throws Exception {
+        return client.listTasks().size();
       }
     };
   }

